central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -3,24 +3,61 @@ package compliancemanager
 import (
 	"context"
 
+	"github.com/pkg/errors"
+	integrationDS "github.com/stackrox/rox/central/complianceoperator/v2/integration/datastore"
 	"github.com/stackrox/rox/central/sensor/service/connection"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/search"
 )
 
 type managerImpl struct {
 	sensorConnMgr connection.Manager
+	integrationDS integrationDS.DataStore
 }
 
 // New returns on instance of Manager interface that provides functionality to process compliance requests and forward them to Sensor.
-func New(sensorConnMgr connection.Manager) Manager {
+func New(sensorConnMgr connection.Manager, integrationDS integrationDS.DataStore) Manager {
 	return &managerImpl{
 		sensorConnMgr: sensorConnMgr,
+		integrationDS: integrationDS,
 	}
 }
 
 func (m *managerImpl) Sync(_ context.Context) {
 	//TODO: Sync scan configurations with sensor
 }
 
+func (m *managerImpl) ProcessComplianceOperatorInfo(ctx context.Context, complianceIntegration *storage.ComplianceIntegration) error {
+	// Check and see if we have this info already.
+	existingIntegrations, err := m.integrationDS.GetComplianceIntegrations(ctx, search.NewQueryBuilder().
+		AddExactMatches(search.ClusterID, complianceIntegration.GetClusterId()).ProtoQuery())
+	if err != nil {
+		return err
+	}
+	// TODO (ROX-18101):  Shouldn't happen once ROX-18101 is implemented.  Deferring more thorough handling
+	// of this condition to that ticket.
+	if len(existingIntegrations) > 1 {
+		return errors.New("multiple compliance operators for this cluster exist.")
```

Comment:
```suggestion
		return errors.Errorf("multiple compliance operators for this cluster %s exist", existingIntegrations[0].GetClusterName())
```
Cluster names are unique in ACS.

Reply:
since storage object does not have the cluster name.  using the cluster id in this log statement.